### PR TITLE
LP-201 Create a new service for Limited Partnerships

### PR DESCRIPTION
**Pull request by HatiDev**
### JIRA link
https://companieshouse.atlassian.net/browse/LP-201

### Change description
Create a new service for Limited Partnerships to allow interaction between Web and API

### Work checklist

- [x] Tests added where applicable
- [ ] UI changes look good on mobile
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/service/limited-partnerships-service.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import { logger } from "../utils/logger";
+
+export const createLimitedPartnership = async (
+
+  transactionId: string,
+  // _data: LimitedPartnership
+): Promise<string> => {
+  logger.info(`Calling 'postLimitedPartnership' for transaction id '${transactionId}'`);
+
+  const result = await createLimitedPartnership(transactionId);
```
[the remark this answers is not in the document]
Comment (markpit) [reply src/service/limited-partnerships-service.ts:10]: Perhaps return a dummy/hardcoded string until the real call is put in place in the future?

[the remark this answers is not in the document]
Comment (HatiDev) [reply src/service/limited-partnerships-service.ts]: Dummy data return added

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/test/limited.partnership.test.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { createLimitedPartnership } from '../service/limited-partnerships-service';
+import { jest } from '@jest/globals';
+
+jest.mock('../service/limited-partnerships-service');
+
+const mockedCreateLP = createLimitedPartnership as jest.MockedFunction<(transactionId: string) => Promise<string>>;
```
[the remark this answers is not in the document]
Comment (HatiDev) [reply src/test/limited.partnership.test.ts]: Not mocking function, mocked the logger instead to still allow for createLimitedPartnership function to be async and use await

File: src/service/limited-partnerships-service.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export const createLimitedPartnership = async (
+
+  transactionId: string,
+
+): Promise<string> => {
+  logger.info(`Calling 'postLimitedPartnership' for transaction id '${transactionId}'`);
+
+  // Return a hardcoded string with await
+  return await Promise.resolve("Dummy response");
```
Comment (mwestacott) [inline src/service/limited-partnerships-service.ts:11]: You can remove the `await` and `Promise.resolve` wrapper code - they cancel each other out. Just return the string.

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/service/limited-partnerships-service.ts (hunk 3 above)
Comment (HatiDev) [reply src/service/limited-partnerships-service.ts]: removed await and Promise wrapper code

**Review by mwestacott — APPROVED** [review]

**Review by markpit — APPROVED** [review]

**Pull request merged**